drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,269 @@
+
+    0x80000003 & 0x7FFFFFFFF = 0x00000003 = 3
+
+Version Group Id
+----------------
+
+The version group id is a non-zero, random and unique identifier assigned to a transaction format version, or a group of soft-forking transaction format versions.  The version group id helps nodes disambiguate between branches using the same version number.
+
+That is, it prevents a client on one branch of the network from attempting to parse transactions intended for another branch, in the situation where the transactions share the same format version number but are actually specified differently.  For example, Zcash and Zclone might both define their own custom v3 transaction formats, but each will have its own unique version group id, so that they can reject v3 transactions with unknown version group ids.
+
+The combination of transaction version and version group id, ``nVersion || nVersionGroupId``, uniquely defines the transaction format, thus enabling parsers to reject transactions from outside the client's chain which cannot be parsed.
+
+By convention, it is expected that when introducing a new transaction version requiring a network upgrade, a new unique version group id will be assigned to that transaction version.
+
+However, if new transaction versions are soft-fork compatible with older transaction versions, the same version group id can be re-used.
+
+Expiry Height
+-------------
+
+The expiry height field, as defined in the Transaction Expiry ZIP [#zip-???]_, stores the block height after which a transaction will be removed from the mempool if it has not been mined.
+
+Transaction Validation
+======================
+
+A valid Overwinter transaction intended for Zcash must have:
+
+- version number 3
+- version group id as specified in Zcash source code
+- overwintered flag set
+
+Overwinter transaction parsers should reject transactions for violating consensus rules if:
+
+- the overwintered flag is not set
+- the version group id is unknown
+- the version number is unknown
+
+Implementation
+==============
+
+The comments and code samples in this section apply to the reference C++ implementation of Zcash.  Other implementations may vary.
+
+Transaction Version
+-------------------
+
+Transaction version remains a positive value.  The main Zcash chain will follow convention and continue to order transaction versions in an ascending order.
+
+Tests can continue to check for the existence of forwards-compatible transaction fields by checking the transaction version using comparison operators::
+
+    if (tx.nVersion >= 2) {
+        for (int js = 0; js < joinsplits; js++) {
+            ...
+        }
+    }
+
+When (de)serializing v3 transactions, the version group id should also be checked in case the transaction is intended for a branch which has a different format for its version 3 transaction::
+
+    if (tx.nVersion == 3 && tx.nVersionGroupID == OVERWINTER_VERSION_GROUP_ID) {
```

Comment:
s/nVersionGroupID/nVersionGroupId/ (note case of 'd')

Reply:
I always find that uppercase `ID` stands out more in code.